code/controllers/subsystem/persistence.dm

```diff
@@ -262,6 +262,19 @@ SUBSYSTEM_DEF(persistence)
 	if(fexists(frame_path))
 		return json_decode(file2text(frame_path))
 
+/// Removes the identifier of a persitent photo frame from the json.
+/datum/controller/subsystem/persistence/proc/RemovePhotoFrame(identifier)
+	var/frame_path = file("data/photo_frames.json")
+	if(!fexists(frame_path))
+		return
+
+	var/frame_json = json_decode(file2text(frame_path))
+	fdel(frame_path)
+	frame_json -= identifier
+
+	frame_json = json_encode(frame_json)
+	WRITE_FILE(frame_path, frame_json)
```

Comment:
only fdel right before you write to the file to make sure you don't lose the data if something runtimes inbetween